lib/spack/spack/relocate_text.py

```diff
@@ -0,0 +1,240 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
```

Comment:
Can we add more docstrings to this module (top-level and in classes / functions)?